src/main/java/io/aiven/elasticsearch/repositories/gcs/GcsRepositoryPlugin.java

```diff
@@ -76,10 +76,15 @@ private GcsBlobStoreRepository createBlobStoreRepository(final RepositoryMetadat
 
     @Override
     public void reload(final Settings settings) {
-        try {
-            gcsSettingsProvider.reload(settings);
-        } catch (final IOException e) {
-            throw new UncheckedIOException(e);
+        // configure plugin only in case we have
+        // settings with prefix aiven
+        final var hasAivenKeys = !settings.filter(key -> key.startsWith("aiven.")).isEmpty();
+        if (hasAivenKeys) {
```

Comment:
Please consider using `settings.getByPrefix`

Reply:
wont work. settings.getPrefix will work for settings like: `bla.bla.bla.{type}` :-) and they need define as `Setting.prefixKeySetting` ... quite from docs:
`
This setting type allows to validate settings that have the same type and a common prefix. For instance feature.${type}=[true|false]
`

Reply:
I added FIXME .. need some time for investigation ... i got failures, when i tried to use `getByPrefix()`

Reply:
BTW, it's supposed to strip away the prefix, may it be the cause?

Reply:
I checked code it adds `.` before filtering.

Reply:
Same here added FIXME